src/app/task-details/task-details.component.ts

```diff
@@ -27,27 +30,48 @@ export class TaskDetailsComponent implements OnInit {
   }
 
   ngOnInit(): void {
-    const taskId = +this.route.snapshot.paramMap.get('id')!;
-    this.task = this.taskService.getTasks()[taskId];
-    if (this.task){
-      this.taskForm.setValue({
-        title: this.task.title,
-        description: this.task.description,
-        type: this.task.type,
-        status: this.task.status
-      });
+    const taskId = this.route.snapshot.paramMap.get('id');
+    if (taskId) {
+      this.getTask(taskId);
     }
+
+  }
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+  getTask(id: string) {
+    this.taskService.getATask(id).pipe(takeUntil(this.destroy$)).subscribe({
+      next: (response: Task) => {
+        this.task = response;
+        if (this.task) {
+          this.taskForm.setValue({
+            title: this.task.title,
+            description: this.task.description,
+            type: this.task.type,
+            status: this.task.status
+          });
+        }
+      },
+      error: (error: Error) => {
+        console.error(error);
+      }
+    })
+
   }
-  editMode(){
+  editMode() {
     this.isEditMode = !this.isEditMode;
   }
-  onSave(){
-    if (this.taskForm.valid && this.task){
-      const updatedTask = {...this.task,...this.taskForm.value};
-      this.taskService.updateTask(updatedTask);
+  onSave() {
+    if (this.taskForm.valid && this.task) {
+      const updatedTask = { ...this.task, ...this.taskForm.value };
```

Comment:
shouldn't the updated task only consist of the taskForm value? As the old and new values will be apart of that form